sympy/calculus/util.py

```diff
@@ -0,0 +1,109 @@
+from sympy import S, sympify, diff, limit, oo
+from sympy.calculus.singularities import singularities
+from sympy.sets.sets import Interval, Intersection, FiniteSet, Union, Complement, Set, EmptySet
+from sympy.solvers.solveset import solveset
+
+
+def not_empty_in(finset_intersection, *syms):
+    """ Finds the domain of the functions in `finite_set` in which the
+    `finite_set` is not-empty
+
+    Parameters
+    ==========
+
+    finset_intersection: The unevaluated intersection of FiniteSet containing
+                        real-valued functions with Union of Sets
+    syms: Tuple of symbols
+            Symbol for which domain is to be found
+
+    Raises
+    ======
+
+    NotImplementedError
+        The algorithms to find the non-emptiness of the given FiniteSet are
+        not yet implemented.
+    ValueError
+        The input is not valid.
+    RuntimeError
+        It is a bug, please report it to the github issue tracker
+        (https://github.com/sympy/sympy/issues).
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, FiniteSet, Interval, not_empty_in, sqrt, oo
+    >>> from sympy.abc import x
+    >>> not_empty_in(FiniteSet(x/2).intersect(Interval(0, 1)), x)
+    [0, 2]
+    >>> not_empty_in(FiniteSet(x, x**2).intersect(Interval(1, 2)), x)
+    [-sqrt(2), -1] U [1, 2]
+    >>> not_empty_in(FiniteSet(x**2/(x + 2)).intersect(Interval(1, oo)), x)
+    (-2, -1] U [2, oo)
+    """
+
+    # TODO: handle piecewise defined functions
+    # TODO: handle transcendental functions
+    # TODO: handle multivariate functions
+    if len(syms) == 0:
+        raise ValueError("A Symbol or a tuple of symbols must be given \
+                            as the third parameter")
+
```

Comment:
You do not need backslash. You can use " instead. Python will handle spaces properly. Currently

```
In [1]: FiniteSet(x/2)
Out[1]: 
⎧x⎫
⎨─⎬
⎩2⎭

In [2]: not_empty_in(_)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-584d108ddade> in <module>()
----> 1 not_empty_in(_)

/home/leosartaj/projects/sympy/sympy/calculus/util.py in not_empty_in(finset_intersection, *syms)
     47     if len(syms) == 0:
     48         raise ValueError("A Symbol or a tuple of symbols must be given \
---> 49                             as the third parameter")
     50 
     51     if finset_intersection.is_EmptySet:

ValueError: A Symbol or a tuple of symbols must be given                             as the third parameter
```